ffi/derive/src/attr_parse/getset.rs

```diff
@@ -0,0 +1,830 @@
+            Self::MutGetters => GetSetGenMode::GetMut,
+            Self::CopyGetters => GetSetGenMode::GetCopy,
+        }
+    }
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Clone)]
+pub struct GetSetOptions {
+    pub visibility: Option<syn2::Visibility>,
+    pub with_prefix: bool,
+}
+
+struct SpannedGetSetOptions {
+    span: Span,
+    options: GetSetOptions,
+}
+
+impl syn2::parse::Parse for SpannedGetSetOptions {
+    fn parse(input: ParseStream) -> syn2::Result<Self> {
+        let mut result = GetSetOptions::default();
+        // an accumulator for syn errors?
+        // this is getting out of hand...
+        // we need an accumulator to rule them all!
+        let mut errors = Vec::new();
+
+        let lit = input.parse::<syn2::LitStr>()?;
+        for part in lit.value().split(' ') {
+            if part == "with_prefix" {
+                result.with_prefix = true;
+            } else if let Ok(vis) = syn2::parse_str::<syn2::Visibility>(part) {
+                if result.visibility.is_none() {
+                    result.visibility = Some(vis);
+                } else {
+                    errors.push(syn2::Error::new(
+                        lit.span(),
+                        format!(
+                            "Failed to parse getset options at {}: duplicate visibility",
+                            part
+                        ),
+                    ));
+                }
+            } else {
+                errors.push(syn2::Error::new(lit.span(), format!("Failed to parse getset options at `{}`: expected visibility or `with_prefix`", part)));
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(SpannedGetSetOptions {
+                span: lit.span(),
+                options: result,
+            })
+        } else {
+            let mut errors = errors.into_iter();
+            let mut error = errors.next().expect("darling::Error can never be empty");
+
+            for next_error in errors {
+                error.combine(next_error);
+            }
+
+            Err(error)
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub enum GetSetGenMode {
+    Get,
+    GetCopy,
+    Set,
+    GetMut,
+}
+
+impl Display for GetSetGenMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            GetSetGenMode::Get => write!(f, "get"),
+            GetSetGenMode::GetCopy => write!(f, "get_copy"),
+            GetSetGenMode::Set => write!(f, "set"),
+            GetSetGenMode::GetMut => write!(f, "get_mut"),
+        }
+    }
+}
+
+impl FromStr for GetSetGenMode {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "get" => Ok(GetSetGenMode::Get),
+            "get_copy" => Ok(GetSetGenMode::GetCopy),
+            "set" => Ok(GetSetGenMode::Set),
+            "get_mut" => Ok(GetSetGenMode::GetMut),
+            _ => Err(()),
+        }
+    }
+}
```

Comment:
Why not [`parse_display`](https://docs.rs/parse-display/latest/parse_display/#enum-format)?

```rust
#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone, parse_display::Display, parse_display::FromStr)]
#[display(style = "snake_case")]
pub enum GetSetGenMode {
    Get,
    GetCopy,
    Set,
    GetMut,
}
```

Reply:
Oh, nice, I've been googling for a proc macro like this, but couldn't find it